asapdiscovery-alchemy/asapdiscovery/alchemy/cli/utils.py

```diff
@@ -259,3 +259,59 @@ def report_alchemize_clusters(alchemical_clusters, outsiders):
         ]
     )
     return alchemical_cluster_sizes, outsider_cluster_sizes, alchemical_num_in_clusters
+
+
+def cinnabar_femap_is_connected(fe_map):
+    """Checks whether the provided femap is connected. Convenience function to make function
+    naming clearer compared to cinnabar nomenclature."""
+    return fe_map.check_weakly_connected()
+
+
+def cinnabar_femap_get_largest_subnetwork(fe_map, result_network, console):
+    """From a disconnected femap, returns the subnetwork with the largest number of nodes using a networkx
+    workaround. Requires the original FreeEnergyCalculationNetwork to query results from.
+
+    Returns a cinnabar FEMap that is fully connected"""
+    import itertools
+
+    import networkx as nx
+    from asapdiscovery.alchemy.schema.fec import (
+        AlchemiscaleResults,
+        FreeEnergyCalculationNetwork,
+    )
+    from rich.padding import Padding
+
+    fe_map_nx = fe_map.graph
+    subnetworks_nodenames = sorted(  # split the network into subnetworks
+        nx.strongly_connected_components(fe_map_nx), key=len, reverse=True
+    )
```

Comment:
If we could just extract the largest subnetwork from the graph we might be able to convert directly back to an FEmap using `from_networkx` which would make this a lot more simple. https://github.com/OpenFreeEnergy/cinnabar/blob/c140fea77d4019119ed40acd1a699b92ed6bbf10/cinnabar/femap.py#L174-L184

Reply:
the commit that added that feature isn't in the conda-installable conda-forge release of `cinnabar`. is there a reasonable way of pinning our `cinnabar` dep to that most recent commit? if it's too hacky I think we might be better off sticking with this approach as it's a bit more controllable than a magic 'from_networkx()` method imo.

Reply:
Good point lets stick with the current method until cinnabar gets updated!